src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -422,6 +422,22 @@ public String getCookie() {
         }
 
         @Override public CauseOfBlockage getCauseOfBlockage() {
+            Run<?, ?> run = runForDisplay();
+            if (!stopping && run != null && !run.isLogUpdated()) {
+                stopping = true;
+                LOGGER.warning(() -> "Refusing to build " + this + " and cancelling it because associated build is complete");
+                Timer.get().execute(() -> {
+                    Queue.getInstance().cancel(this);
+                });
```

Comment:
```suggestion
                Timer.get().execute(() -> Queue.getInstance().cancel(this));
```

Reply:
Ah yeah, at one point I was logging something in the block and I forgot to simplify it.